decoder.go

```diff
@@ -280,69 +413,107 @@ func (d *Decoder) Reset() {
 
 	d.err = nil
 	d.transformComplete = false
-}
+
+	d.debug1 = false
+	d.debugSpam = false
+	if d.segId != nil {
+		// Reset the segment ID to an empty string
+		// to prevent nil pointer dereference.
+		// If a new segment ID is needed, it should be set via SetSegmentId.
+		d.segId = new(string)
+	}
+} // end func Reset
 
 func (d *Decoder) processYenc(line []byte) {
-	if bytes.HasPrefix(line, []byte("=ybegin ")) {
-		d.begin = true
-		d.m.Size, _ = extractInt(line, []byte(" size="))
-		d.m.Name, _ = extractString(line, []byte(" name="))
-		if _, err := extractInt(line, []byte(" part=")); err != nil {
+	dlog(d.debugSpam, "rapidyenc.processYenc(%q)", line)
+	switch d.format {
+	case FormatYenc:
```

Comment:
I'd suggesting splitting this into `processYenc` and `processUU` and revert the checks around L314 because to be FormatYenc we already know the line must have started `=y`